modules/gui/include/visp3/gui/vpPclPointCloudVisualization.h

```diff
@@ -0,0 +1,252 @@
+ *
+ * For using ViSP with software that can not be combined with the GNU
+ * GPL, please contact Inria about acquiring a ViSP Professional
+ * Edition License.
+ *
+ * See http://visp.inria.fr for more information.
+ *
+ * This software was developed at:
+ * Inria Rennes - Bretagne Atlantique
+ * Campus Universitaire de Beaulieu
+ * 35042 Rennes Cedex
+ * France
+ *
+ * If you have questions regarding the use of this file, please contact
+ * Inria at visp@inria.fr
+ *
+ * This file is provided AS IS with NO WARRANTY OF ANY KIND, INCLUDING THE
+ * WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
+ * 
+ * 
+ *
+ * Description:
+ * Real-time 3D point clouds plotter based on the PCL library.
+ *
+ * Authors:
+ * Romain LAGNEAU
+ *
+ *****************************************************************************/
+
+#ifndef POINTCLOUDVISUALIZATION_H
+#define POINTCLOUDVISUALIZATION_H
+
+#ifndef DOXYGEN_SHOULD_SKIP_THIS
+
+#include <visp3/core/vpConfig.h>
+
+#if defined(VISP_HAVE_PCL)
+// System
+#include <thread>
+#include <mutex>
+
+// ViSP
+#include <visp3/core/vpColVector.h>
+
+// PCL
+#include <pcl/visualization/pcl_visualizer.h>
+
+class VISP_EXPORT vpPclPointCloudVisualization
+{
+public:
+  typedef typename pcl::PointXYZRGB pclPoint;
+  typedef typename pcl::PointCloud<pclPoint> pclPointCloud;
+  typedef typename pclPointCloud::Ptr pclPointCloudPtr;
+
+  /**
+   * @brief Structure that contains all the required parameters to display a legend on the visualizer.
+   */
+  typedef struct legendParams{
+    std::string _text; /*!< The text of the legend.*/
+    unsigned int _pos_u; /*!< The position of the legend on the u-axis of the screen.*/
+    unsigned int _pos_v; /*!< The position of the legend on the v-axis of the screen.*/
+    unsigned int _size; /*!< The size of the legend.*/
+    double _rRatio; /*!< The red ratio of the legend.*/
+    double _gRatio; /*!< The green ratio of the legend.*/
+    double _bRatio; /*!< The blue ratio of the legend.*/
+  }legendParams;
+
+  /**
+   * @brief Construct a new vpPclPointCloudVisualization object.
+   * 
+   * @param title The title of the window of the visualizer.
+   * @param width The width of the window of the visualizer.
+   * @param height The height of the window of the visualizer.
+   * @param px The position on the horizontal axis of the screen of the window of the visualizer.
+   * @param py The position on the vertical axis of the screen of the window of the visualizer.
+   * @param outFolder If different from the empty string, the point clouds will be saved in this folder.
+   * @param ignoreThreshold A point for which the weight is below this threshold will be displayed in black.
+   */
+  vpPclPointCloudVisualization(const std::string &title , const int &width = 640, const int &height = 480, const int &px = 720, const int &py = 560, const std::string &outFolder = std::string(), const double &ignoreThreshold = 0.95);
+  ~vpPclPointCloudVisualization();
```

Comment:
There should be the same with `vpCameraParameters` as parameters (to replace `px` and `py`

Reply:
The names were confusing, px and py designated the position of the window on the screen.
They have been renamed posU and posV in [041a1de](https://github.com/lagadic/visp/pull/1163/commits/041a1dea7b13fe767606ba4736180b136765dbe8)